src/app/common/transactions/generate-unsigned-txs.ts

```diff
@@ -50,7 +50,7 @@ function generateUnsignedContractCallTx(args: GenerateUnsignedContractCallTxArgs
     contractAddress,
     functionName,
     publicKey,
-    anchorMode: AnchorMode.Any,
+    anchorMode: txData.anchorMode ?? AnchorMode.Any,
```

Comment:
```suggestion
    anchorMode: anchorMode ?? AnchorMode.Any,
```

Just to keep this inline with everything else, we could destructure as above

Reply:
Good shout, updated :)